yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -464,7 +472,7 @@ private[spark] class ApplicationMaster(args: ApplicationMasterArguments,
   /**
    * Actor that communicates with the driver in client deploy mode.
```

Comment:
i donot think it, when yarn client mode, it need to transfer false to AMActor. because AMActor and Driver do not  belong to same process. so AMActor  need to keep a long connect with Driver. when Driver exit, AMActor need to detect it. but in yarn cluster mode, because AMActor and driver belong to same process, they do not need to keep a long connect.